met-web/src/components/engagement/view/widgets/Poll/PollWidgetView.tsx

```diff
@@ -0,0 +1,175 @@
+
+const PollWidgetView = ({ widget }: PollWidgetViewProps) => {
+    const dispatch = useAppDispatch();
+    const [pollWidget, setPollWidget] = useState<PollWidget>({
+        id: 0,
+        widget_id: 0,
+        engagement_id: 0,
+        title: '',
+        description: '',
+        status: '',
+        answers: [],
+    });
+
+    const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
+    const [isLoading, setIsLoading] = useState(true);
+    const [selectedOption, setSelectedOption] = useState('');
+    const [isSubmitted, setIsSubmitted] = useState(false);
+    const [interactionEnabled, setInteractionEnabled] = useState(true);
+    const [responseMessage, setResponseMessage] = useState(RESPONSE_MESSAGE_SUCCESS);
+    const { getSubmittedPolls, addSubmittedPoll } = useSubmittedPolls();
+
+    useEffect(() => {
+        // Check if the current widget ID is in the submitted polls
+        if (getSubmittedPolls().includes(widget.id)) {
+            setIsSubmitted(true);
+        }
+        fetchPollDetails();
+    }, [widget]);
+
+    const fetchPollDetails = async () => {
+        try {
+            const poll_widgets = await fetchPollWidgets(widget.id);
+            const poll_widget = poll_widgets[poll_widgets.length - 1];
+            setPollWidget(poll_widget);
+            setIsLoading(false);
+        } catch (error) {
+            setIsLoading(false);
+            console.log(error);
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Error occurred while fetching Engagement widgets information',
+                }),
+            );
+        }
+    };
+    // Type guard for HttpResponseError
+    const isHttpResponseError = (error: unknown): error is HttpResponseError => {
+        return error instanceof Error && 'response' in error;
+    };
+
+    const handleSubmit = async () => {
+        if (selectedOption == '') {
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'You need to select an answer to the poll before submitting it.',
+                }),
+            );
+            return;
+        }
+        setInteractionEnabled(false);
+        try {
+            await postPollResponse(widget.id, pollWidget.id, {
+                selected_answer_id: parseInt(selectedOption),
+            });
+        } catch (error: unknown) {
+            const responseMessage =
+                isHttpResponseError(error) && error.response?.status === 403
+                    ? RESPONSE_MESSAGE_LIMIT
+                    : RESPONSE_MESSAGE_ERROR;
+            setResponseMessage(responseMessage);
+        } finally {
+            setInteractionEnabled(true);
+        }
+        // Irrespective of the error mark the poll as submitted to avoid re-submitting
+        setIsSubmitted(true);
```

Comment:
Yes, its for anti-spamming. But as you pointed out we need to consider server unavailability scenario too. Will update the logic like if limit error disable poll otherwise keep it enabled for re-submission